src/array/cuda/sddmm.cu

```diff
@@ -91,6 +91,43 @@ void SDDMMCsr(const std::string& op,
   });
 }
 
+/*!
+ * \brief CUDA implementation of g-SDDMM on heterograph using 
+    Csr format.
+ */
+template <int XPU, typename IdType, int bits>
```

Comment:
Does bits need to be a template parameter? Can we just parse it from `vec_lhs`, `vec_rhs`, `vec_out` or `vec_csr`?

Reply:
Sounds good.